test/contractBridge/whitelist.js

```diff
@@ -0,0 +1,139 @@
+        assert.isTrue(await BridgeInstance.hasRole(ADMIN_ROLE, adminAccount));
+        await TruffleAssert.passes(BridgeInstance.enableWhitelist({from: adminAccount}))
+        await TruffleAssert.passes(BridgeInstance.disableWhitelist({from: adminAccount}))
+
+        assert.isFalse(await BridgeInstance.hasRole(WHITELISTER_ROLE, genericAccount));
+        await TruffleAssert.reverts(BridgeInstance.enableWhitelist({from: genericAccount}),
+            "Sender doesn't have Whitelister or Admin role")
+        await TruffleAssert.reverts(BridgeInstance.disableWhitelist({from: genericAccount}),
+            "Sender doesn't have Whitelister or Admin role")
+
+        await BridgeInstance.grantRole(WHITELISTER_ROLE, genericAccount);
+        await TruffleAssert.passes(BridgeInstance.enableWhitelist({from: genericAccount}))
+        await TruffleAssert.passes(BridgeInstance.disableWhitelist({from: genericAccount}))
+    });
+
+    it('should require admin or whitelist role to operate whitelist', async () => {
+        assert.isTrue(await BridgeInstance.hasRole(ADMIN_ROLE, adminAccount));
+        await TruffleAssert.passes(BridgeInstance.addToWhitelist(testAccount, {from: adminAccount}))
+        await TruffleAssert.passes(BridgeInstance.removeFromWhitelist(testAccount, {from: adminAccount}))
+
+        assert.isFalse(await BridgeInstance.hasRole(WHITELISTER_ROLE, genericAccount));
+        await TruffleAssert.reverts(BridgeInstance.addToWhitelist(testAccount, {from: genericAccount}),
+            "Sender doesn't have Whitelister or Admin role")
+        await TruffleAssert.reverts(BridgeInstance.removeFromWhitelist(testAccount, {from: genericAccount}),
+            "Sender doesn't have Whitelister or Admin role")
+
+        await BridgeInstance.grantRole(WHITELISTER_ROLE, genericAccount);
+        await TruffleAssert.passes(BridgeInstance.addToWhitelist(testAccount, {from: genericAccount}))
+        await TruffleAssert.passes(BridgeInstance.removeFromWhitelist(testAccount, {from: genericAccount}))
+    });
+
+    it('should revert if whitelist is already enabled/disabled', async () => {
+        await TruffleAssert.reverts(BridgeInstance.disableWhitelist(),
+            "Whitelist is already disabled")
+        await BridgeInstance.enableWhitelist();
+        await TruffleAssert.reverts(BridgeInstance.enableWhitelist(),
+            "Whitelist is already enabled")
+    });
+
+    it('should revert if whitelisted address already added/removed', async () => {
+        await TruffleAssert.reverts(BridgeInstance.removeFromWhitelist(testAccount),
+            "Address to remove is not whitelisted")
+        await BridgeInstance.addToWhitelist(testAccount);
+        await TruffleAssert.reverts(BridgeInstance.addToWhitelist(testAccount),
+            "Address to add is already whitelisted")
+    });
+
+    it('should apply whitelisting by adding custom modifier to a method', async () => {
```

Comment:
Split this unit test into a few. It looks too heavy.

Reply:
[done](https://github.com/Cerebellum-Network/chainbridge-solidity/pull/4/files/b21035db27edc18a3d62142a3e7f7389eb4804f6#diff-a62c2a6bda93583bed6bafc173326ced2d611a6ae8a37740d97001a613dceefeR100)